{
    "name": "numpy",
    "cve_id": "CVE-2021-33430",
    "publish_date": "2021-12-17T19:15Z",
    "cwe_id": "CWE-120",
    "owner": "Travis E. Oliphant et al.",
    "affected_versions": [
        {
            "version_value": "1.10.0",
            "operator": "<"
        }
    ],
    "references": [
        "https://nvd.nist.gov/vuln/detail/CVE-2021-33430",
        "https://github.com/numpy/numpy/issues/18939",
        "https://github.com/numpy/numpy/commit/ae317fd9ff3e79c0eac357d723bfc29cbd625f2e",
        "https://github.com/advisories/GHSA-6p56-wp2h-9hxr"
    ],
    "description": "From NVD:\nA Buffer Overflow vulnerability exists in NumPy 1.9.x in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service.\nFrom Github:\nA Buffer Overflow vulnerability exists in NumPy 1.9.x in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service.",
    "latest_version": "1.22.0",
    "language": "python",
    "license": "BSD-3-Clause",
    "repository_summary": "NumPy is the fundamental package for array computing with Python.",
    "repo_url": "https://github.com/numpy/numpy",
    "impact": {
        "cvss2_score": 5.0,
        "cvss2_severity": "MEDIUM",
        "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
        "av": "NETWORK",
        "ac": "LOW",
        "pr": "NONE",
        "c": "NONE",
        "i": "NONE",
        "a": "HIGH",
        "impact_score": 3.6,
        "exploitability_score": 3.9,
        "cvss3_score": 7.5,
        "cvss3_severity": "HIGH",
        "ui": "NONE",
        "s": "UNCHANGED"
    },
    "ochrona_severity_score": 7.5
}